### ISSUE TITLE
Race condition in update

### ISSUE THREAD

**Issue by leedm777** [balderdashy/sails-mysql#189]
There is a race condition in `update` (in `adapter.js`) that I've run into a few times. The current update implementation works by:

1. Query with the given criteria to get a list of ids
2. Update with the given criteria
3. Query with the list of ids from step 1 to get the updated objects

The problem is that if other updated happen between steps 1 and 2, the data returned in step 3 could be inaccurate (either omitting/including records that were/were not affected).

Since this is a MySQL specific driver, there's a [pretty clever way](http://stackoverflow.com/a/1751282/115478) of capturing the ids for updated records.

**Comment by devinivy** [comment]
Good idea–would love to see a PR for this.

**Comment by sailsbot** [comment]
Thanks for posting, @leedm777.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
